geomstats/algebra_utils.py

```diff
@@ -114,8 +114,12 @@ def from_vector_to_diagonal_matrix(vector, num_diag=0):
         is a diagonal matrix containing the `i`-th row of `vector`.
     """
     num_columns = gs.shape(vector)[-1]
-    identity = gs.eye(num_columns)
-    identity = gs.cast(identity, vector.dtype)
+    identity = gs.repeat(
+        gs.zeros_like(gs.reshape(vector, (-1, num_columns))[:1, :]),
+        repeats=num_columns,
+        axis=0,
+    )
+    identity[range(num_columns), range(num_columns)] = 1
```

Comment:
`identity = gs.eye(num_cols, dtype=vector.dtype` suffices, no?

Reply:
No, it is not sufficient for two reasons: 
- the shape of the output would be 2D, we want it to be 2D or 3D. 
- the `device` would always be CPU.

Reply:
See above.